crates/bevy_asset/src/assets.rs

```diff
@@ -69,13 +87,26 @@ impl<T: Asset> Assets<T> {
         self.get_handle(id)
     }
 
+    /// Add/modify the asset pointed to by the given handle.
+    ///
+    /// You __must__ use the returned Strong handle. Failing to do so may result in the unexpected release of the asset.
+    ///
+    /// See [set_untracked](Assets::set_untracked) for more info.
     #[must_use = "not using the returned strong handle may result in the unexpected release of the asset"]
     pub fn set<H: Into<HandleId>>(&mut self, handle: H, asset: T) -> Handle<T> {
         let id: HandleId = handle.into();
         self.set_untracked(id, asset);
         self.get_handle(id)
     }
 
+    /// Add/modify the asset pointed to by the given handle.
+    ///
+    /// If an asset already exists with the given HandleId, it will be modified. Otherwise the new asset
+    /// will be inserted.
+    ///
+    /// # Events
+    /// * [`AssetEvent::Created`]: Sent if the asset did not yet exist with the given handle
+    /// * [`AssetEvent::Modified`]: Sent if the asset with given handle already existed
```

Comment:
Is this worthwhile to put in the documentation? Personally, I'd like to know what methods trigger certain events, but I'm not too sure how everyone else feels about adding something like this. Let me know if it should be removed/modified!

Reply:
looks useful to me 👍